UART_driver/Core/Inc/CircularBuffer.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef CIRCULAR_BUFFER_HPP
+#define CIRCULAR_BUFFER_HPP
+
+#include <stdint.h>
+
+class CircularBuffer {
+    public:
+        // Constructor. `buf` is the buffer to be used as a circular buffer
```

Comment:
I know this hasen't been documented yet, but try to follow the doxygen style comments shown in the image below: I beleive if you use vscode and have like intellisense or something, and you start typing it out it would recognize it and fill it in for you... nevertheless when you write comments like this is really lays things out, makes it readable and it shows up nice when you hover over the function when you have some extensions in vs code.

<img width="532" alt="image" src="https://github.com/UWARG/firmware-prototypes/assets/69175242/bd250ac6-754f-4b93-88ce-9d0a54d6ba37">